docs/snippets/vue/button-story-default-docs-code.js.mdx

```diff
@@ -0,0 +1,37 @@
+```js
+// Button.stories.js
+
+import Button from './Button.vue';
+
+export default {
+  title: 'Button',
+  component: Button,
+  argTypes: {
+    backgroundColor: { control: "color" }
+  },
+};
+
+const someFunction = someValue => {
+  return `i am a ${someValue}`;
+};
+
+export const ExampleStory = (args, { argTypes }) => {
+  // destructure the label from the args object
+  const oldArgs = args;
+  const { label } = oldArgs;
+  //
+  // assigns the function result to a variable and pass it as a prop into the component
+  const functionResult = someFunction(label);
+  args.label = functionResult;
```

Comment:
Does this work?

Reply:
@shilman code is [here](https://github.com/jonniebigodes/storybook-docs-vue-snippets) and deployed Storybook [here](https://flamboyant-perlman-0eb03f.netlify.app/?path=/story/example-introduction--page). It's arranged to match the docs TOC. Regarding your comment i understand where you come from to not encourage the pattern, but it's one of those cases where we might get "bitten" by not providing examples and with that leading the vue users that it's not possible to it with Vue. I'm more than willing to make a compromise here and see what @tmeasday has to say about it.

Reply:
This is the example located in [here](https://storybook.js.org/docs/react/writing-docs/doc-blocks) for React:

```js
/import React from 'react';

import { Button } from './Button';

// some function to demonstrate the behavior
const someFunction = (someValue) => {
  return `i am a ${someValue}`;
};

export default {
  title: 'Button',
  component: Button,
  argTypes: {
    backgroundColor: { control: 'color' }
  },
};

export const ExampleStory = (args) => {
  // destructure the label from the args object
  const { label } = args;
  //
  // assigns the function result to a variable and pass it as a prop into the component
  const functionResult = someFunction(label);
  return <Button {...args} label={functionResult} />;
};

ExampleStory.args = {
  primary: true,
  size: 'small',
  label: 'button'
};
```

In a nutshell we're using one of the args provided in the story as a function argument and use the result of said function as the value that is going into the prop `label`. To give a little context this was asked sometime last year and i created this simple example. Now when this is applied to Vue things get a bit interesting as how Storybook is setup for Vue. I've tinkered with this and came up with 3 options:
1- it's the one used here
2- I went with the following:
```js
export const ExampleStory = (args, { argTypes }) => {
  const { label } = args;
  const functionResult = someFunction(label);
  return {
    props: Object.keys(argTypes),
    components: { Button },
   template: `<Button label="${functionResult}" v-bind="$props"/>`
  };
};
// Same code(args) as provided in this snippet 
```
Using this reactivity is lost, basically it works once, but when me or the reader goes tries to update the label arg in the controls panel it will not update.
3- As a test (i knew it wouldn't actually work but went with it for testing purposes):
```js
export const ExampleStory = (args, { argTypes }) => {
  const { label } = args;
  const functionResult = someFunction(label);
  return {
    props: Object.keys(argTypes),
    components: { Button },
    template: `<Button v-bind="$props" v-bind:label="${functionResult}"/>`
  };
};
// Same code(args) as provided in this snippet 
```
Will throw an error.

I know that this is not the ideal approach for Vue (and Angular) but if anyone has a better idea on how to apply this to Vue i'm willing to update the snippet and get this merged. 

Feel free to provide feedback